src/utils/useProjectId.ts

```diff
@@ -1,25 +1,54 @@
 import { useChannel } from "@storybook/manager-api";
 import React from "react";
 
-import { UPDATE_PROJECT, UpdateProjectPayload } from "../constants";
+import {
+  PROJECT_UPDATED,
+  PROJECT_UPDATING_FAILED,
+  ProjectUpdatingFailedPayload,
+  UPDATE_PROJECT,
+  UpdateProjectPayload,
+} from "../constants";
 
 const { CHROMATIC_PROJECT_ID } = process.env;
 
 export const useProjectId = (): [
   projectId: string,
+  projectToken: string,
+  configDir: string,
   updateProject: (projectId: string, projectToken?: string) => void,
-  projectIdChanged: boolean,
-  clearProjectIdChanged: () => void
+  projectUpdatingFailed: boolean,
+  projectIdUpdated: boolean,
+  clearProjectIdUpdated: () => void
 ] => {
   const [projectId, setProjectId] = React.useState<string | null>(CHROMATIC_PROJECT_ID);
-  const [projectIdChanged, setProjectIdChanged] = React.useState(false);
+  const [projectToken, setProjectToken] = React.useState<string | null>();
+  const [projectIdUpdated, setProjectIdUpdated] = React.useState(false);
+  const [projectUpdatingFailed, setProjectUpdatingFailed] = React.useState(false);
+  const [configDir, setConfigDir] = React.useState<string | null>();
 
-  const emit = useChannel({});
+  const emit = useChannel({
+    [PROJECT_UPDATED]: () => setProjectIdUpdated(true),
+    [PROJECT_UPDATING_FAILED]: (payload: ProjectUpdatingFailedPayload) => {
+      setProjectUpdatingFailed(true);
+      setConfigDir(payload.configDir);
+    },
+  });
 
-  const updateProject = (newProjectId: string, projectToken: string) => {
-    emit(UPDATE_PROJECT, { projectId: newProjectId, projectToken } as UpdateProjectPayload);
+  const updateProject = (newProjectId: string, newProjectToken: string) => {
+    emit(UPDATE_PROJECT, {
+      projectId: newProjectId,
+      projectToken: newProjectToken,
+    } as UpdateProjectPayload);
     setProjectId(newProjectId);
-    setProjectIdChanged(true);
+    setProjectToken(newProjectToken);
   };
-  return [projectId, updateProject, projectIdChanged, () => setProjectIdChanged(false)];
+  return [
+    projectId,
+    projectToken,
+    configDir,
+    updateProject,
+    projectUpdatingFailed,
```

Comment:
This was renamed to updatingProjectFailed when used. We might want to change this to an object to get name consistency. We can do that later.

Reply:
Good pickup, thanks.